internal/fdo.go

```diff
@@ -64,9 +64,57 @@ func fdoLookupApplication(theme string, size int, appName string) desktop.IconDa
 			return newFdoIconData(theme, size, testLocation)
 		}
 	}
+	//No Match Was Found Checking File Names - Check Files Themselves
+	for _, dataDir := range locationLookup {
+		testLocation := filepath.Join(dataDir, "applications")
+		files, err := ioutil.ReadDir(testLocation)
+		if err != nil {
+			continue
+		}
+		for _, f := range files {
+			if strings.HasPrefix(f.Name(), ".") || f.IsDir() {
+				continue
+			}
+			icon := newFdoIconData(theme, size, filepath.Join(testLocation, f.Name()))
+			if icon != nil {
+				if icon.name == appName {
+					return icon
+				} else if icon.exec == appName {
+					return icon
+				}
+			}
+		}
+	}
 	return nil
 }
 
+//fdoLookupApplicationPartial looks up an application by a partial name and returns all matches in an fdoIconData struct slice
+func fdoLookupApplicationPartialMatch(theme string, size int, appName string) []desktop.IconData {
```

Comment:
This should be plural as it returns many applications.
I wonder if just "Matching" rather than "PartialMatch" is clearer?